### PR TITLE
Major refactor of acf.py

### PR DESCRIPTION
**Pull request by RemingtonRohel**
# Scope 

* Changed all classmethods to staticmethods
* Created staticmethod plot_acf_param() to handle plotting of values, which is called within plot_acfs() as needed to plot real, imaginary, power, and phase of ACF
* Return all axes from plot_acfs() as list
* Plot real and imaginary ACF values on input ax to plot_acfs() if pwr_and_phs False, otherwise use plt.gca()
* Removed some unnecessary internal variables in plot_acfs().
* Proper handling of markers for blanked lags, taking into consideration if marker is a filled marker or not

**issue:** N/A

## Approval

**Number of approvals:** *1*

## Test

**matplotlib version**: 3.7.2

Try running the tutorial from https://pydarn.readthedocs.io/en/main/user/acf/, and using as many of the optional arguments as you can.

### CODE REVIEW

**Comment by RemingtonRohel** [comment]
Made another commit as I realized I had inadvertently changed the phase plotting to plot in radians. It is now back to plotting in degrees.

File: pydarn/plotting/acf.py [hunk 1]
```diff
@@ -253,104 +250,89 @@ def plot_acfs(cls, dmap_data: List[dict], beam_num: int = 0,
...
+        ax.set_ylabel(parameter)
+        ax.set_xlabel('Lag Number')
+        real_dict['data']['line'].set_label('Real')
+        imag_dict['data']['line'].set_label('Imag')
+        if plot_blank:
+            real_dict['data']['blank_line'].set_label('Real Blanked')
+            imag_dict['data']['blank_line'].set_label('Imag Blanked')
+            lim_val = 1.1 * max(np.nanmax(np.abs(re)), np.nanmax(np.abs(im)))
+        else:
+            lim_val = 1.1 * max(np.nanmax(np.abs(masked_re), np.nanmax(np.abs(masked_im))))
```
Comment (carleyjmartin) [inline pydarn/plotting/acf.py:283]: This line throws a: `TypeError: 'numpy.float64' object cannot be interpreted as an integer`
when I use plot_blank=False

:(

```
rtn = pydarn.ACF.plot_acfs(data, beam_num=5, gate_num=12, normalized=True,
                start_time=datetime(2018, 5, 20, 9, 0), pwr_and_phs=True,
                plot_blank=False)
```

Comment (carleyjmartin) [reply pydarn/plotting/acf.py:283]: Also sorry I've tried to figure out why it's doing this but I dunno

**Review by carleyjmartin — COMMENTED** [review]
Other than the tiny error, looks great for everything else.

File: pydarn/plotting/acf.py (hunk 1 above)
Comment (carleyjmartin) [reply pydarn/plotting/acf.py:283]:
```suggestion
            lim_val = 1.1 * max(np.nanmax(np.abs(masked_re),
                                            np.nanmax(np.abs(masked_im))))
```

And the line is too long style wise 👀

Comment (RemingtonRohel) [reply pydarn/plotting/acf.py]: Found and fixed the bug! Trying to uses nanmax on a masked array is not a good idea, by default masked_array.max() ignores masked values and all the NaN's were already masked.

**Pull request merged**